Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -126,6 +135,77 @@ namespace AzToolsFramework
             return true;
         }
 
+        bool InstanceDomGenerator::GenerateEntityDom(PrefabDom& entityDom, const AZ::Entity& entity) const
+        {
+            // Retrieves the focused instance.
+            auto prefabFocusInterface = AZ::Interface<PrefabFocusInterface>::Get();
+            if (!prefabFocusInterface)
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "PrefabFocusInterface could not be found.");
+                return false;
+            }
+
+            InstanceOptionalReference focusedInstanceRef = prefabFocusInterface->GetFocusedPrefabInstance(s_editorEntityContextId);
+            if (!focusedInstanceRef.has_value())
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "Could not get the focused instance. It should not be null.");
+                return false;
+            }
+            const Instance& focusedInstance = focusedInstanceRef->get();
+            const Instance* focusedInstancePtr = &focusedInstance;
+
+            const AZ::EntityId entityId = entity.GetId();
+            InstanceOptionalReference owningInstanceRef = m_instanceEntityMapperInterface->FindOwningInstance(entityId);
+            if (!owningInstanceRef.has_value())
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "Could not get the owning instance for the given entity id.");
+                return false;
+            }
+            const Instance& owningInstance = owningInstanceRef->get();
+
+            // Climbs up from the owning instance to root instance, but stops at the focused instance if they can meet.
+            const InstanceClimbUpResult climbUpResult = PrefabInstanceUtils::ClimbUpToTargetOrRootInstance(owningInstance, focusedInstancePtr);
+            const Instance* focusedOrRootInstancePtr = climbUpResult.m_reachedInstance;
+            if (!focusedOrRootInstancePtr)
+            {
+                AZ_Assert(false, "Prefab - InstanceDomGenerator::GenerateEntityDom - "
+                    "Could not get the focused or root instance from climb-up. It should not be null.");
+                return false;
+            }
+
+            // Prepares the path from the focused (or root) instance to the entity.
+            AZStd::string relativePathFromTop = PrefabInstanceUtils::GetRelativePathFromClimbedInstances(climbUpResult.m_climbedInstances);
+            AZStd::string entityAliasPath = m_instanceToTemplateInterface->GenerateEntityAliasPath(entityId);
+            AZStd::string relativePathFromTopToEntity = relativePathFromTop + entityAliasPath;
```

Comment:
You can just append to relativePathFromTop to avoid string copy here since its a temp variable anyway.